src/ResetPasswordHelper.php

```diff
@@ -154,6 +154,25 @@ public function getTokenLifetime(): int
         return $this->resetRequestLifetime;
     }
 
+    /**
+     * Generate a fake reset token.
+     *
+     * Use this to generate a fake token so that you can, for example, show a
+     * "reset confirmation email sent" page that includes a valid "expiration date",
+     * even if the email was not actually found (and so, a true ResetPasswordToken
+     * was not actually created).
+     *
+     * This method should not be used when timing attacks are a concern.
```

Comment:
Why not make this method as *close* as possible to the real one, to mitigate timing? What Specifically, I'm thinking about running: https://github.com/SymfonyCasts/reset-password-bundle/blob/6825b9f30de14fa28c679082096373989cd138f2/src/ResetPasswordHelper.php#L75 .. even if we then don't use it.

Reply:
that would help, but we would need to init a user entity. 

what about changing the method signature to `public function generateFakeResetToken(object $user = null): ResetPasswordToken` then `if null !== $user { $this-repo->getUserIdentifier()` ?

This would allow you to generate fake tokens w/o needing persistence. But to slim down the timing attack vector, a consumer would have to init and supply an entity if they choose to (not something we could / should do in the bundle itself)

Reply:
Ah, good point. I vote for... let's leave it for later if someone wants it.